unit-test/validate-bump.bats

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env ./libs/bats/bin/bats
+
+# SAMPLE outline
+# @test "Test name" {
+#   run *command*
+#   True or False based on exit status of the script
+# }
+
+@test "Nightly : dev" {
```

Comment:
For your test names, we generally follow a pattern where the name is something like "should (desired behavior) when (test condition)". You can see plenty of examples in the test files in this repo, like [here](https://github.com/iTwin/itwinjs-core/blob/f507b7ae8d916fee56a1b84a3fed54ae72b590e8/core/backend/src/test/IModelHost.test.ts#L89) for example. That way, when you see a failing test, just based on the test name, you know what failed, what should have happened, and the conditions for the test. For example, this test checks to see if validation passes when you are trying to apply a nightly version bump and the current version is a nightly. In this instance, you could name the test something like "should pass validation when nightly version bump is applied to a dev version"